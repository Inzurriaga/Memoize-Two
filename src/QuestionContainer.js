import React, { Component } from "react"
import questions from "./MockData"

export default class QuestionContainer extends Component{
    questionSelected = (question) => {
        this.props.toggleToQuestion(question)
    }
    render(){
        return(
            <ul className="questionContainer-section">
                {
                    questions.map((question) => {
                        if(this.props.user.completed.includes(question.id)){
                            return(
                                <li className="user-completed" onClick={() => { this.questionSelected(question) }}
                                     key={question.id}>{question.title}</li>
                            )     
                        }else{
                            return(
                                <li className="user-notCompleted" onClick={() => { this.questionSelected(question) }}
                                     key={question.id}>{question.title}</li>
                            )
                        }
        
                    })
                }
            </ul>
        )
    }
}