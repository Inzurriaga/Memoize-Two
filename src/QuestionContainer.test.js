import React from 'react';
import { shallow } from 'enzyme';
import QuestionContainer from "./QuestionContainer"

const mockUser = {"name":"gabriel","password":"gabriel","completed":[1,2,5]};
const mockToggleToQuestion = jest.fn();
const mockSelectedQuestion = { title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1 };

describe("QuestionContainer", ()=> {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <QuestionContainer questionSelected={mockSelectedQuestion}
                          user={mockUser}
                          toggleToQuestion={mockToggleToQuestion}/>
    )
  })
  it("should match Snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should invoke question when li element is click", () => {
    wrapper.find("li").at(0).simulate("click")
    expect(mockToggleToQuestion).toHaveBeenCalled()
  })

  it("should show the li as green if player already answer the question", () => {
    let mockUser = {"name":"gabriel","password":"gabriel","completed":[]};
    wrapper = shallow(
      <QuestionContainer questionSelected={mockSelectedQuestion}
                          user={mockUser}
                          toggleToQuestion={mockToggleToQuestion}/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})