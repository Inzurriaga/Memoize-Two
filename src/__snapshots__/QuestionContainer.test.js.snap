// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionContainer should match Snap shot 1`] = `
<ul
  className="questionContainer-section"
>
  <li
    className="user-completed"
    key="1"
    onClick={[Function]}
  >
    number return 5
  </li>
  <li
    className="user-completed"
    key="2"
    onClick={[Function]}
  >
    return hello world
  </li>
  <li
    className="user-notCompleted"
    key="4"
    onClick={[Function]}
  >
    return an array
  </li>
  <li
    className="user-completed"
    key="5"
    onClick={[Function]}
  >
    number return a 3
  </li>
</ul>
`;

exports[`QuestionContainer should show the li as green if player already answer the question 1`] = `
<ul
  className="questionContainer-section"
>
  <li
    className="user-notCompleted"
    key="1"
    onClick={[Function]}
  >
    number return 5
  </li>
  <li
    className="user-notCompleted"
    key="2"
    onClick={[Function]}
  >
    return hello world
  </li>
  <li
    className="user-notCompleted"
    key="4"
    onClick={[Function]}
  >
    return an array
  </li>
  <li
    className="user-notCompleted"
    key="5"
    onClick={[Function]}
  >
    number return a 3
  </li>
</ul>
`;
